packages/platform-sdk-ark/src/services/ledger.ts

```diff
@@ -41,4 +58,42 @@ export class LedgerService implements Contracts.LedgerService {
 	public async signMessage(path: string, payload: Buffer): Promise<string> {
 		return this.#transport.signMessageWithSchnorr(path, payload);
 	}
+
+	public async scan(): Promise<Contracts.WalletData[]> {
+		const pageSize = 5;
+		let page = 0;
+		const slip44 = this.#config.get<number>("network.crypto.slip44");
+
+		let wallets: Contracts.WalletData[] = [];
+		let collection: WalletDataCollection;
+		do {
+			const addresses: string[] = [];
+
+			for (const index of createRange(page, pageSize)) {
+				const path: string = formatLedgerDerivationPath({ coinType: slip44, account: index });
+				const publicKey = await this.getPublicKey(path);
+				addresses.push(await this.#identity.address().fromPublicKey(publicKey));
+			}
+
+			collection = await this.#client.wallets({ addresses: addresses });
+			wallets = wallets.concat(collection.items());
+
+			page++;
+		} while (!collection.isEmpty());
```

Comment:
We stop when all the generated addresses are cold wallets. Is this fine? Any other stopping criteria?

Reply:
Should be fine since that is how the Desktop Wallet works if I remember right.